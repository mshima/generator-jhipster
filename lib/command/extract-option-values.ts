import type { Simplify } from 'type-fest';
import type { ExtractOptionValuesfromConfigs, JHipsterConfigs } from './types.js';
import { upperFirst } from 'lodash-es';

export const extractOptionValuesFromConfig = <const Configs extends JHipsterConfigs>(
  configs: Configs,
): Simplify<ExtractOptionValuesfromConfigs<Configs>> => {
  const withChoices = Object.entries(configs)
  .filter(([_name, config]) => config.choices);
  return {
    ...Object.fromEntries(withChoices.map(([name]) => [`${name}Option`, name])),
    ...Object.fromEntries(withChoices.map(([name, config]) => config.choices!.map(choice => (typeof choice === 'string' ? choice : choice.value)).map(choice =>[
        `${name}Value${upperFirst(choice)}`,
        choice,
      ])).flat(),
  )
  } as any;
};
