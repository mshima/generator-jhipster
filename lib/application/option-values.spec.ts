import { expect } from 'esmocha';
import { lookupCommandsConfigs } from '../command/lookup-commands-configs.js';
import { extractOptionValuesFromConfig } from '../command/extract-option-values.js';
import optionValues from './option-values.js';

describe('OptionValues', () => {
  it('should match snapshot', () => {
    expect(optionValues).toMatchSnapshot();
  });
  it('should contain every option', async () => {
    const configs = await lookupCommandsConfigs();
    expect(optionValues).toMatchObject(extractOptionValuesFromConfig(configs));
  });
});
