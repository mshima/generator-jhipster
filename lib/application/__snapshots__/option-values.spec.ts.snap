// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionValues should match snapshot 1`] = `
{
  "applicationTypeOption": "applicationType",
  "applicationTypeValueGateway": "gateway",
  "applicationTypeValueMicroservice": "microservice",
  "applicationTypeValueMonolith": "monolith",
  "authenticationTypeOption": "authenticationType",
  "authenticationTypeValueJwt": "jwt",
  "authenticationTypeValueOauth2": "oauth2",
  "authenticationTypeValueSession": "session",
  "databaseMigrationOption": "databaseMigration",
  "databaseMigrationValueLiquibase": "liquibase",
  "databaseTypeOption": "databaseType",
  "databaseTypeValueCassandra": "cassandra",
  "databaseTypeValueCouchbase": "couchbase",
  "databaseTypeValueMongodb": "mongodb",
  "databaseTypeValueNeo4j": "neo4j",
  "databaseTypeValueNo": "no",
  "databaseTypeValueSql": "sql",
  "defaultPackagingOption": "defaultPackaging",
  "defaultPackagingValueJar": "jar",
  "defaultPackagingValueWar": "war",
  "messageBrokerOption": "messageBroker",
  "messageBrokerValueKafka": "kafka",
  "messageBrokerValueNo": "no",
  "messageBrokerValuePulsar": "pulsar",
  "serviceDiscoveryTypeOption": "serviceDiscoveryType",
  "serviceDiscoveryTypeValueConsul": "consul",
  "serviceDiscoveryTypeValueEureka": "eureka",
  "serviceDiscoveryTypeValueNo": "no",
}
`;
